network/src/consensus.rs

```diff
@@ -152,24 +152,35 @@ impl<P, E> Network for ConsensusNetwork<P,E> where
 
 		// spin up a task in the background that processes all incoming statements
 		// TODO: propagate statements on a timer?
-		let inner_stream = self.network.consensus_gossip().write().messages_for(attestation_topic);
-		let process_task = self.network
-			.with_spec(|spec, ctx| {
+		let (tx, rx) = std::sync::mpsc::channel();
+		self.network.with_gossip(move |gossip, _| {
+			let inner_rx = gossip.messages_for(attestation_topic);
+			let _ = tx.send(inner_rx);
+		});
+
+		let table_router_clone = table_router.clone();
+		let executor = task_executor.clone();
+
+		// spin up a task in the background that processes all incoming statements
+		// TODO: propagate statements on a timer?
+		self.network
+			.with_spec(move |spec, ctx| {
```

Comment:
this whole block could be moved up a level of indentation if we didn't do a newline on `.with_spec`